src/SpeedyTransforms/scratch_memory.jl

```diff
@@ -0,0 +1,50 @@
+"""
+ScratchMemory holds scratch memory for the `SpectralTransform` that's used both by the Fourier and Legendre transform. Fields are
+$(TYPEDFIELDS)"""
+struct ScratchMemory{
+    NF,
+    ArrayComplexType,           # <: ArrayType{Complex{NF}, 3},
+}
+    # SCRATCH MEMORY FOR FOURIER NOT YET LEGENDRE TRANSFORMED AND VICE VERSA
+    # state is undetermined, only read after writing to it
+    north::ArrayComplexType
+    south::ArrayComplexType
```

Comment:
Currently only `north` and `south` are here because only those parts of the scratch memory are used in function calls which is what counts for Enzyme (apparently). 

I actually also tried having all parts of the scratch memory in here, but I had some problems with it that I couldn't quite figure out and doing it like it is now with just `north` and `south` also means that `_fourier!` and `_legendre!` don't need to be changed at all.